### ISSUE TITLE
javascript in a react-script-editor seems to run when republishing a modern SharePoint page and no other time

### ISSUE THREAD

**Issue by timleveritt** [pnp/sp-dev-fx-webparts#1235]
## Category

- [x] Question
- [ ] Bug
- [ ] Enhancement
- [ ] Sample request

## Authors

@mikaelsvenson

## Expected or Desired Behavior

Javascript in the React-Script-Editor web part of a widget to a [JobAdder](https://jobadder.com/) Job Board renders open jobs for applicants to view and apply on a Modern SharePoint page in a Communication site.

## Observed Behavior

A javascript widget is currently working as expected on a Classic team site (also the tenant root site) and webpart page.
The same javascript in the React-Script-Editor webpart in a modern page on a Communication site renders the Job Board only after editing and republishing the modern page, not on first save of the page nor when just navigating to or refreshing the page.
The sample @mikaelsvenson provided to the TradingView widget works fine so to me the web part seems to be ok.

So the question is why would the JobAdder widget work in Classic but partially (and effectively not at all) in Modern?

I'm happy to share the javascript but not publicly.

## Submission Guidelines

**Comment by hugoabernier** [comment]
@timleveritt, I don't believe that this issue is related to the web part itself, but most likely due to the page lifecycle of modern pages.

Modern pages don't always fully render when you refresh or navigate to another page, but use partial rendering instead in order to improve page performance. In such cases, if the widget is relying on a traditional `DOMCntentLoaded` or `window.load` event, your widget will not be triggered, as these events may not occur.

I'll go ahead and close this issue, but if you still need help, please don't hesitate to open a new issue with additional information so that we can help you better.
